semeio/workflows/ahm_analysis/ahmanalysis.py

```diff
@@ -0,0 +1,465 @@
+    grid_param = Grid(input_grid.rsplit(".", 1)[0], fformat="eclipserun")
+    all_input = []
+    for file_path in files:
+        proproff = GridProperty(file_path, name=param_name, grid=grid_param)
+        array_nb = proproff.get_npvalues1d(activeonly=False, fill_value=0, order="C")
+        all_input.append(array_nb)
+    return all_input
+
+
+def _export_field_param(config_node, file_system, ensemble_size, output_path):
+    # Get/export the updated Field parameters
+    EnkfNode.exportMany(
+        config_node,
+        output_path,
+        file_system,
+        np.arange(0, ensemble_size),
+    )
+
+
+def make_update_log_df(update_log_dir):
+    """ Read update_log file to get active and inactive observations """
+    list_of_files = [
+        os.path.join(update_log_dir, f) for f in os.listdir(update_log_dir)
+    ]
+    if len(list_of_files) > 1:
+        raise OSError("ERROR more than one update_log_file.")
+    if len(list_of_files) == 0:
+        raise OSError("ERROR empty update_log directory.")
+    # read file
+    updatelog = pd.read_csv(
+        list_of_files[0],
+        delim_whitespace=True,
+        skiprows=6,
+        usecols=[0, 2, 3, 5, 6, 8, 10],
+        header=None,
+        skipfooter=1,
+    )
+    # define header
+    updatelog.columns = [
+        "obs_number",
+        "obs_key",
+        "obs_mean",
+        "obs_std",
+        "status",
+        "sim_mean",
+        "sim_std",
+    ]
+
+    # ---------------------------------
+    # add proper name for obs_key in rows where value is '...'
+    updatelog.replace("...", np.nan, inplace=True)
+    updatelog.ffill(inplace=True)
+
+    return updatelog
+
+
+def make_obs_groups(key_map):
+    """Create a mapping of observation groups, the names will be:
+    data_key -> [obs_keys] and All_obs-{missing_obs} -> [obs_keys]
+    and All_obs -> [all_obs_keys]
+    """
+    combinations = key_map.copy()
+    if len(combinations) == 1:
+        return combinations
+    # We want all observations and all observations
+    # minus individual observations:
+    for combination_length in (len(key_map.keys()) - 1, len(key_map.keys())):
+        if combination_length == 1:
+            # We have hit a case with two observation groups, now we only
+            # need individual group and all obs.
+            continue
+        for subset in itertools.combinations(key_map.keys(), combination_length):
+            obs_group = list(
+                itertools.chain.from_iterable([key_map[key] for key in subset])
+            )
+            if combination_length == len(key_map.keys()) - 1:
+                missing_obs = [x for x in key_map.keys() if x not in set(subset)]
+                assert len(missing_obs) == 1
+                name = f"All_obs-{missing_obs[0]}"
+            else:
+                name = "All_obs"
+            combinations[name.replace(":", "_")] = obs_group
+    return combinations
+
+
+def get_nb_active_observations(df_update_log):
```

Comment:
I would suggest renaming this to `count_active_observations`

Reply:
ok done